packages/components/src/higher-order/with-filters/index.tsx

```diff
@@ -1,7 +1,7 @@
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { Component, WPComponent } from '@wordpress/element';
```

Comment:
Could this fix the static linting CI error?

```suggestion
import type { Component, WPComponent } from '@wordpress/element';
```

Reply:
Mhh, I tried running `npm run lint:js` on my machine but I couldn't reproduce the CI error.

From the docs, it looks like the [`import/named` rule](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/named.md) should be disabled for TypeScript configs, but for some reason it's erroring here.

Maybe we can force-disable the rule?

```suggestion
// eslint-disable-next-line import/named
import { Component, WPComponent } from '@wordpress/element';
```

Reply:
What the... switched out with `React.ComponentType` like you suggested 👍